kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -773,4 +895,70 @@ private void addMetricsReporters(Properties props) {
   public RestMetricsContext getMetricsContext() {
     return metricsContext.metricsContext();
   }
+
+  private final class ConfigsBuilder {
+    private final Set<String> mask ;
+    private final Map<String, ConfigValue> configs = new HashMap<>();
+
+    private ConfigsBuilder(Set<String> mask) {
+      this.mask = requireNonNull(mask);
+    }
+
+    private ConfigsBuilder addConfigs(String prefix, boolean strip) {
+      Map<String, ConfigValue> toAdd =
+          Maps.filterKeys(originalsWithPrefix(prefix, strip), mask::contains)
+              .entrySet()
+              .stream()
+              .collect(
+                  Collectors.toMap(
+                      Entry::getKey,
+                      entry ->
+                          ConfigValue.create(
+                              (strip ? prefix : "") + entry.getKey(), entry.getValue())));
```

Comment:
this is probably right, i can't be bothered thinking about it, but just want to note that when strip is true, you're adding the prefix, which seems backwards, just in case you've made a mistake.

Reply:
The first argument, `origin`, tracks the original configuration name, as specified by the user. So if we did strip the prefix (strip = true), we have to re-append it to get the original name back. It looks a little backwards, I know, but it's doing the right thing haha.